### PR TITLE
Update documentation for rename

### LINKED ISSUE #13473: ENH: add errors='raise' option to rename
Is there a reason that `.rename` doesn't raise if keys are supplied that don't exist? This caught me out a couple of times:

``` python
In [14]: import pandas as pd

In [15]: df=pd.DataFrame({'a': [1,2], 'b': [3,4]})

In [16]: df
Out[16]: 
   a  b
0  1  3
1  2  4

In [17]: df.rename(columns={'a': 'c'})
Out[17]: 
   c  b
0  1  3
1  2  4
# ok

In [18]: df.rename(columns={'d': 'c'})
Out[18]: 
   a  b
0  1  3
1  2  4

# this should raise?
```

### PR DESCRIPTION
**Pull request by gliptak**
- [x] closes #13473
- [ ] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
needs an update to the doc string as well

**Comment by gliptak** [comment]
Updated docstring

File: pandas/core/generic.py [hunk 1]
```diff
@@ -555,8 +555,8 @@ def swaplevel(self, i=-2, j=-1, axis=0):
     _shared_docs['rename'] = """
         Alter axes input function or functions. Function / dict values must be
         unique (1-to-1). Labels not contained in a dict / Series will be left
-        as-is. Alternatively, change ``Series.name`` with a scalar
-        value (Series only).
+        as-is. Extra labels listed don't throw an error. Alternatively, change
```
Comment (jreback) [inline pandas/core/generic.py]: can you add 1 or more examples in Examples

File: doc/source/basics.rst [hunk 2]
```diff
@@ -1159,14 +1159,17 @@ mapping (a dict or Series) or an arbitrary function.
...
-Series, it need only contain a subset of the labels as keys:
+labels (and must produce a set of unique values). A dict or
+Series can also be used:
 
 .. ipython:: python
 
    df.rename(columns={'one' : 'foo', 'two' : 'bar'},
              index={'a' : 'apple', 'b' : 'banana', 'd' : 'durian'})
 
+If mapping doesn't name a column/index, it isn't renamed. Also extra names
```
Comment (jorisvandenbossche) [inline doc/source/basics.rst]: "If mapping doesn't name a column/index," -> "If the mapping doesn't include a column/index label"

**Comment by jorisvandenbossche** [comment]
@gliptak A small grammatical comment, otherwise looks good!

File: pandas/core/generic.py (hunk 1 above)
Comment (gliptak) [inline pandas/core/generic.py]: added

File: doc/source/basics.rst (hunk 2 above)
Comment (gliptak) [inline doc/source/basics.rst]: thank you. corrected

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**